packages/file_selector/file_selector_android/android/src/main/java/dev/flutter/packages/file_selector_android/FileUtils.java

```diff
@@ -130,11 +135,13 @@ public static String getPathFromCopyOfFileFromUri(@NonNull Context context, @Non
         fileName = getBaseName(fileName) + extension;
       }
 
-      File file = new File(targetDirectory, fileName);
+      String filePath = new File(targetDirectory, fileName).getPath();
+      File outputFile = saferOpenFile(filePath, targetDirectory.getCanonicalPath());
```

Comment:
This now throw an exception. Is there more work we need to do to turn that illegal argument exception into something a dart developer can handle?

Reply:
When I was primarily an app/plugin dev we had a rule that plugins couldn't throw exceptions that are not handled in dart. Flutter/packages is not as strict but I would like us to investigate what it would take to let the host app know that this error happened. Keep in mind that the original report required a malicious content provider so there could be reasonable app responses to this that help the user not just prevent the bad behavior. Like letting an app developer know that someone is trying to use this attack to hurt their users. Or telling a user that the content provider they selected might be malicious.

Reply:
I've passed the Java exception through to dart and thrown there, and exported the type